filament/src/details/Material.cpp

```diff
@@ -1140,13 +1140,30 @@ void FMaterial::precacheDepthVariants(FEngine& engine) {
 void FMaterial::processDescriptorSets(FEngine& engine, MaterialParser const* const parser) {
     UTILS_UNUSED_IN_RELEASE bool success;
 
+
     success = parser->getDescriptorBindings(&mProgramDescriptorBindings);
     assert_invariant(success);
 
     std::array<backend::DescriptorSetLayout, 2> descriptorSetLayout;
     success = parser->getDescriptorSetLayout(&descriptorSetLayout);
     assert_invariant(success);
 
+    // Setup Labels for debugging
+    utils::CString namestr;
+    utils::StaticString perViewStr = "_perView";
+    utils::StaticString singleStr = "_single";
+
+    parser->getName(&namestr);
+
+    utils::CString singleLabel(namestr.c_str(), namestr.length() + singleStr.length());
+    memccpy(singleLabel.c_str() + namestr.length(), singleStr.c_str(), '\0', singleStr.length());
+
+    utils::CString perViewLabel(namestr.c_str(), namestr.length() + perViewStr.length());
+    memccpy(perViewLabel.c_str() + namestr.length(), perViewStr.c_str(), '\0', perViewStr.length());
+
+    descriptorSetLayout[0].label = singleLabel;
+    descriptorSetLayout[1].label = perViewLabel;
```

Comment:
I think if you rebased on top of main, the perView layouts are moved out of Material.